src/platforms/common/server/owning_c_str.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIR_OWNING_C_STR_
+#define MIR_OWNING_C_STR_
+#include <cstdlib>
+#include <memory>
+namespace mir
+{
+using CStr = std::unique_ptr<char[], decltype([](char const* p) { ::free(const_cast<char *>(p)); })>;
```

Comment:
Huh. You learn something new every day. `decltype(<lambda literal>)` is a neat way of getting a type with a specific `operator()`!